### ISSUE TITLE
Encoding of instants and intervals

### ISSUE THREAD

**Issue by cportele** [opengeospatial/ogc-feat-geo-json#41]
The JSON encoding of instants and intervals is very close to their encoding in [CQL2 JSON](https://docs.ogc.org/DRAFTS/21-065.html), but there are small differences. Should we use the same encoding as CQL2?

A comparison:

## A date

CQL2 uses a more specific key:

**JSON-FG**

```json
{ "instant": "1969-07-20" }
```

**CQL2**

```json
{ "date": "1969-07-20" }
```

## A timestamp

CQL2 uses a more specific key:

**JSON-FG**

```json
{ "instant": "1969-07-20T20:17:40Z" }
```

**CQL2**

```json
{ "timestamp": "1969-07-20T20:17:40Z" }
```

## An interval

Identical:

**JSON-FG**

```json
{ "interval": [ "1969-07-16T05:32:00Z", "1969-07-24T16:50:35Z" ] }
```

**CQL2**

```json
{ "interval": [ "1969-07-16T05:32:00Z", "1969-07-24T16:50:35Z" ] }
```

## A half-bounded interval

CQL2 uses the ".." notation from ISO 8601, JSON-FG uses `null`:

**JSON-FG**

```json
{ "interval": [ "2014-04-24T10:50:18Z", null ] }
```

**CQL2**

```json
{ "interval": [ "2014-04-24T10:50:18Z", ".." ] }
```

**Comment by pvretano** [comment]
@cportele yes, it would make sense to be consistent with the nomenclature and conventions so JSON-FG should be updated to match CQL2.

**Comment by hylkevds** [comment]
Having a date in an instant field seems very wrong to me:
```{ "instant": "1969-07-20" }```
A date is not an instant, but an interval, spanning that entire day. 

This also ties back into #40, since as soon as you actually start _working_ with a date, you also run into time zone problems, since a day in one time zone only partially overlaps with a day in another time zone, and the date format does not include time zone information. A plain date could span 50 hours...

The advantage of using a separate field name for `date` is that you can then state that 
```{ "instant": "1969-07-20" }```
is a shorthand for
```{ "interval": "1969-07-20T00:00:00Z", "1969-07-21T00:00:00Z" }```

**Comment by pvretano** [comment]
@hylkevds doesn't the interpretation of an instant depend on the granularity?  If the temporal granularity is a DAY then 1969-07-20 is an instant ... no?  Perhaps we need to introduce the concept of granularity?  Just thinking out loud ...

**Comment by hylkevds** [comment]
That doesn't solve the time zone problem.
Also, going into granularity comes down to stating that all times are an interval and instants don't exist...

**Comment by hylkevds** [comment]
I see three ways to go about the date issue:
1. Granularity is a thing and instants don't exist. So `{ "date": "1969-07-20" }` is just a shorthand for `{ "interval": "1969-07-20T00:00:00Z", "1969-07-21T00:00:00Z" }`
2. Instants do exist, and giving a date just means _leaving off the trailing zeros_. In that case `{ "date": "1969-07-20" }` is a shorthand for `{ "instant": "1969-07-20T00:00:00.000Z" }`In this case the date can live in the `"instant"` member, though putting it in the `"date"` member makes parsing more reliable.
3. Instants do exist, but a _date_ is defined as a shorthand for the interval spanning the day. In this case the date really should go in a `"date"` field to not cause confusion.

The second option has the problem that it puts the instant right at the start of the day, meaning that if it is displayed in another time zone, it has a 50% chance of rolling back to the previous day. Not what users expect. One option to fix that might to be define the date-instance at noon, but that probably has its own issues.

The third option is probable the most pragmatic solution and what most users would expect. Though time zones can still cause confusion.

A wild idea to solve the time zone issue for dates is to add a time zone field next to the date.

**Comment by cportele** [comment]
My understanding is the following:

* We use [OWL Time](https://www.w3.org/TR/owl-time/) as the conceptual basis.
* Instants do exist and Date and Timestamp are different data types to describe the temporal position. What is used will depend on the underlying data. Typically, if a date is used that indicates that the granularity is days and software that uses the data will process the data accordingly (and not cast the date to a timestamp or interval).
* Maybe we should provide guidance in JSON-FG how to cast a date to a timestamp (or interval), where this is necessary, but I am not convinced that we should. We don't provide such guidance for any other data types.

**Comment by KathiSchleidt** [comment]
just confirming @hylkevds worries - while probably less relevant when just dealing with spatial features, we've had great fun dealing with the confusion of a daily set of hourly measurements for Europe ending up encoded in UTC from 23-23, whereby the relevant date is the 2nd day of this interval.
Thus - word to the wise - be careful on this (PLEASE!)

**Comment by cportele** [comment]
Meeting 2022-02-21: We agreed to align with the CQL2 JSON encoding.

On the topic about dates and their interpretation, we identified that CQL2 needs to discuss this, because this is important in comparisons involving casts, e.g., from a date to a timestamp. Once this is added in CQL2, we can reference the text from JSON-FG. Date and Timestamp are different data types .

Add OWL Time to the bibliography.

**Comment by cportele** [comment]
The original issue has been resolved by #42.

Additional text has been added about instants. We will add a link to CQL2 when that document includes more discussion about type casts from one type to another (including dates, timestamps and intervals).
